packages/analytics-core/src/analytics.js

```diff
@@ -62,27 +54,63 @@ export default class AvAnalytics {
   }
 
   handleEvent = event => {
-    if (
-      isModifiedEvent(event) ||
-      (event.type === 'click' && !isLeftClickEvent(event)) ||
-      !isValidEventTypeOnTarget(event)
-    ) {
+    if (this.invalidEvent(event)) {
       return;
     }
+
     const target = event.target || event.srcElement;
-    const analyticAttrs = getAnalyticAttrs(target);
+    const { path } = event;
+    let analyticAttrs = {};
+
+    if (this.recursive) {
+      // Reverse the array so we pull attributes from top down
+      path.reverse().forEach(pth => {
+        const attrs = this.getAnalyticAttrs(pth);
+
+        analyticAttrs = { ...analyticAttrs, ...attrs };
+
+        analyticAttrs.action = analyticAttrs.action || event.type;
+      });
+    } else {
+      analyticAttrs = this.getAnalyticAttrs(target);
+      analyticAttrs.elemId =
```

Comment:
Could this part (lines 76-80) be needed for recursive as well? I notice 80 and 72 are the same so maybe move after the else?

Reply:
So the element id may not be the exact element id of the first element that contains the `data-analytics` attributes. We could just say grab the last element that has attrs and use that has the clicked element id.